### DISCUSSION TITLE
Extra id column being retrieve in findAll()

### DISCUSSION THREAD

**Discussion by fabriciofrontarolli-uplandsoftware** [sequelize/sequelize#15596] [Help & Questions] [answered]
I have the following models

/* OrderStatus */
const OrderStatus = sequelize.define('OrderStatus', {
  id: { type: DataTypes.UUID, allowNull: false, primaryKey: true },
  description: { type: DataTypes.STRING }
}, { sequelize, tableName: 'order_status', modelName: 'order_status', freezeTableName: true, timestamps: false });

/* Order */
const Order = sequelize.define('Order', {
  id: { type: DataTypes.UUID, allowNull: false, primaryKey: true },
  order_status_id: { type: DataTypes.UUID },
}, { sequelize, tableName: 'order', modelName: 'order', freezeTableName: true, timestamps: false });

// Associations
OrderStatus.belongsTo(Order, {
  foreignKey: { name: 'order_status_id', type: DataTypes.UUID }
});

Order.hasOne(OrderStatus, {
  foreignKey: { name: 'order_status_id', type: DataTypes.UUID }
});

When I query OrderStatus like so

OrderStatus.findAll({});

It's trying to fetch an extra column in the generated SQL and throwing an error

{ error: column "order_status_id" does not exist

sql:
      'SELECT "id", "description", "order_status_id" FROM "order_status" AS "OrderStatus";',

Why is it trying to fetch this extra column 'order_status_id'?

**Comment by ephys** [comment, marked as answer]
Sequelize adds a primary key by default. Here is the v6 page that explains it and how to remove it https://sequelize.org/docs/v6/other-topics/legacy/#primary-keys

Or if you use sequelize 7: https://sequelize.org/docs/v7/core-concepts/model-basics/#prevent-creating-a-default-pk-attribute
